db/db_impl_compaction_flush.cc

```diff
@@ -1603,6 +1641,28 @@ Status DBImpl::BackgroundCompaction(bool* made_progress,
 
     // Clear Instrument
     ThreadStatusUtil::ResetThreadStatus();
+  } else if (env_->GetBackgroundThreads(Env::Priority::BOTTOM) > 0 &&
```

Comment:
Can you put this Env function call as the last condition? There is uncertainty about how expensive this call can be, because Env is pluggable. If we only check it if it is a full compaction in universal, we call it much infrequently.